### PR TITLE
Add perl-IPC-Cmd to CentOS6 docker container

### PR DESCRIPTION
**Pull request by jdekonin**
* [skip ci]
* added for openssl3 support
* eclipse-openj9/openj9#13686

Signed-off-by: Joe deKoning <joe_dekoning@ca.ibm.com>

### CODE REVIEW

**Comment by jdekonin** [comment]
@pshipton I have modified the `Build-Jenkins-Agent-Container` to reference my fork with this single change.
https://openj9-jenkins.osuosl.org/view/Infrastructure/job/Build-Jenkins-Agent-Container/17/console

When this (successfully) completes, if you can update your PR (14900??) with this container reference change and respin that PR job for x86 we can see what the next failure point is (hopefully none).  **`16` changes to `17` for both JDK8 and JDK11.**
https://github.com/eclipse-openj9/openj9/blob/10bf2cde0eb595f3d808985f0d42699d12efea9a/buildenv/jenkins/variables/defaults.yml#L292

fyi @AdamBrousseau

**Comment by pshipton** [comment]
Looks like it failed again due to a certificate problem. Let me know when a new container is available and I'll test it.

**Comment by jdekonin** [comment]
RIght, seems there are plenty of certificates expired.  Now on build 21. :(
https://openj9-jenkins.osuosl.org/view/Infrastructure/job/Build-Jenkins-Agent-Container/21/console

**Comment by AdamBrousseau** [comment]
In build 21 `./autogen.sh` seems to fail instantly, not sure why.

**Comment by jdekonin** [comment]
jenkins build docker x86 centos6

**Comment by jdekonin** [comment]
@pshipton based on work @vsebe performed previously for redhat6 containers.

https://openj9-jenkins.osuosl.org/job/Build-Jenkins-Agent-Container/24/console
eclipseopenj9/jenkins-agent-x86-centos6:24

**Comment by pshipton** [comment]
We should test centos7 since it's affected by the change, and ubuntu16 since the wget change is common. I'd assume if ubuntu16 works then higher versions will as well. Not sure the platform matters.

**Comment by pshipton** [comment]
@jdekonin the PR build failed
```
01:12:24  Step 1/28 : FROM nvidia/cuda:9.0-devel-ubuntu16.04 AS cuda-dev
01:12:35  Get "https://registry-1.docker.io/v2/": net/http: TLS handshake timeout
```

**Comment by pshipton** [comment]
Testing of eclipseopenj9/jenkins-agent-x86-centos6:24 to build openssl 3.0.7
https://openj9-jenkins.osuosl.org/job/Pipeline-Build-Test-Personal/285

**Comment by pshipton** [comment]
eclipseopenj9/jenkins-agent-x86-centos6:24 is working fine to build openssl 3.

**Comment by AdamBrousseau** [comment]
Build may have that handshake fail because the OS is too old and doesn't have an updated cert store? I seem to recall a Let's Encrypt change a while back that broke curl on old OSs because they don't support new cert protocol or something like that.

**Comment by jdekonin** [comment]
jenkins build docker x86 ubuntu20

**Comment by jdekonin** [comment]
centos7 - Build-Jenkins-Agent-Container/25 - passes the creation of the container but fails uploading to docker hub
ubuntu16 - Build-Jenkins-Agent-Container/26 - fails with what appears to be missing repos...not related to what I modified.  Maybe just out of service.  Not sure we care really as we shouldn't be running this release any more anyways since its out of support.
ubuntu18 - Build-Jenkins-Agent-Container/27 - passes the creation of the container but fails uploading to docker hub

To add onto my todo list would be fixing any possible cert issues.  Doesn't need to block this PR or use of `eclipseopenj9/jenkins-agent-x86-centos6:24` though.

**Comment by jdekonin** [comment]
jenkins build docker x86 centos7

**Comment by pshipton** [comment]
Pls create another PR, or modify this one, to update defaults.yml to use `eclipseopenj9/jenkins-agent-x86-centos6:24`.

File: buildenv/docker/mkdocker.sh [hunk 1]
```diff
@@ -75,7 +75,7 @@ userid=1000
 version=unspecified
 
 # Frequently used commands.
-wget_O="wget --progress=dot:mega -O"
+wget_O="wget --no-check-certificate --progress=dot:mega -O"
```
Comment (keithc-ca) [inline buildenv/docker/mkdocker.sh:78]: I suggest we should only use `--no-check-certificate` where it's needed, i.e. centos:6.

File: buildenv/docker/mkdocker.sh [hunk 2]
```diff
@@ -372,6 +372,7 @@ fi
   echo "    perl-devel \\"
   echo "    perl-ExtUtils-MakeMaker \\" # required by git
   echo "    perl-GD \\"
+  echo "    perl-IPC-Cmd \\"
```
Comment (keithc-ca) [inline buildenv/docker/mkdocker.sh:375]: Please explain why this is needed and what it helps. Is it needed only for centos:6?

File: buildenv/docker/mkdocker.sh [hunk 3]
```diff
@@ -459,7 +460,6 @@ if [ $arch = x86_64 ] ; then
   echo " && $wget_O gettext.tar.gz http://ftp.gnu.org/gnu/gettext/gettext-$gettext_version.tar.gz \\"
   echo " && tar -xzf gettext.tar.gz \\"
   echo " && cd gettext-$gettext_version \\"
-  echo " && ./autogen.sh --skip-gnulib \\"
```
Comment (keithc-ca) [inline buildenv/docker/mkdocker.sh:462]: Please explain how it's safe to skip this step. Again, is it only relevant for centos:6?

File: buildenv/docker/mkdocker.sh (hunk 2 above)
Comment (pshipton) [reply buildenv/docker/mkdocker.sh:375]: We can't compile openssl v3 without this, see infrastructure/issues/6790 for more details.

Comment (pshipton) [reply buildenv/docker/mkdocker.sh:375]: It's needed for all OSes where builds occur. OpenJ9 only builds xlinux on centos:6.

Comment (keithc-ca) [reply buildenv/docker/mkdocker.sh:375]: It would have helped if that fact were mentioned here.

File: buildenv/docker/mkdocker.sh (hunk 1 above)
Comment (jdekonin) [reply buildenv/docker/mkdocker.sh]: I've added an overwrite of wget_O in the case of centos6 && x86_64 only.

File: buildenv/docker/mkdocker.sh (hunk 2 above)
Comment (jdekonin) [reply buildenv/docker/mkdocker.sh]: Comment descriptor added in latest push.

File: buildenv/docker/mkdocker.sh (hunk 3 above)
Comment (jdekonin) [reply buildenv/docker/mkdocker.sh:462]: This step fails on the latest centos6, I don't recall all the details at this point, but removing didn't seem to cause problems with building the component.

File: buildenv/docker/mkdocker.sh [hunk 4]
```diff
@@ -372,6 +372,7 @@ fi
   echo "    perl-devel \\"
   echo "    perl-ExtUtils-MakeMaker \\" # required by git
   echo "    perl-GD \\"
+  echo "    perl-IPC-Cmd \\" # required for openssl v3 compiles
```
Comment (keithc-ca) [inline buildenv/docker/mkdocker.sh:375]: If I remember correctly, it was required to compile openssl 1.1.1s, so the "v3" qualification should be removed.

File: buildenv/docker/mkdocker.sh [hunk 5]
```diff
@@ -459,7 +460,9 @@ if [ $arch = x86_64 ] ; then
   echo " && $wget_O gettext.tar.gz http://ftp.gnu.org/gnu/gettext/gettext-$gettext_version.tar.gz \\"
   echo " && tar -xzf gettext.tar.gz \\"
   echo " && cd gettext-$gettext_version \\"
+if [ $dist$version != centos6 ] ; then
```
Comment (keithc-ca) [inline buildenv/docker/mkdocker.sh:463]: This is in a context that only applies for `$dist = centos`; the test can be simplified to `$version != 6`.

File: buildenv/docker/mkdocker.sh [hunk 6]
```diff
@@ -916,6 +919,10 @@ fi
 }
 
 main() {
+if [ $dist$version = centos6 -a $arch = x86_64 ] ; then
+   # no cert check was needed and verified required for centos6 x86_64
+   wget_O="wget --no-check-certificate --progress=dot:mega -O"
+fi
```
Comment (keithc-ca) [inline buildenv/docker/mkdocker.sh:925]: Maybe a better place for this is after line 171 that applies only to centos6:
```
          if [ $arch = x86_64 ] ; then
            # Certificates are old on CentOS:6 so we can't expect wget to check.
            wget_O="wget --progress=dot:mega --no-check-certificate -O"
          fi
```

**Review by keithc-ca — CHANGES_REQUESTED** [review]

File: buildenv/docker/mkdocker.sh (hunk 4 above)
Comment (pshipton) [reply buildenv/docker/mkdocker.sh:375]: We're already building 1.1.1s and it's working with the 'eclipseopenj9/jenkins-agent-x86-centos6:16' container, so it must not be needed.

File: buildenv/docker/mkdocker.sh (hunk 5 above)
Comment (jdekonin) [reply buildenv/docker/mkdocker.sh]: Update made.

File: buildenv/docker/mkdocker.sh (hunk 6 above)
Comment (jdekonin) [reply buildenv/docker/mkdocker.sh]: Update made.

File: buildenv/docker/mkdocker.sh [hunk 7]
```diff
@@ -459,7 +464,9 @@ if [ $arch = x86_64 ] ; then
   echo " && $wget_O gettext.tar.gz http://ftp.gnu.org/gnu/gettext/gettext-$gettext_version.tar.gz \\"
   echo " && tar -xzf gettext.tar.gz \\"
   echo " && cd gettext-$gettext_version \\"
+if [ $$version != 6 ] ; then
```
Comment (keithc-ca) [inline buildenv/docker/mkdocker.sh:467]: Only one `$` is wanted here.

File: buildenv/docker/mkdocker.sh (hunk 4 above)
Comment (keithc-ca) [reply buildenv/docker/mkdocker.sh:379]: What versions do we build in a docker container? I couldn't even run `configure` for jdk17 successfully in `eclipseopenj9/jenkins-agent-x86-centos6:16`:
```
checking for version string... 17.0.6-internal+0-adhoc..jdk17
configure: Found potential Boot JDK using configure arguments
configure: Potential Boot JDK found at /home/jenkins/bootjdks/jdk16 is not a working JDK; ignoring
configure: Output from java -version was: /home/jenkins/bootjdks/jdk16/bin/java: /lib64/libc.so.6: version `GLIBC_2.14' not found (required by /home/jenkins/bootjdks/jdk16/bin/../lib/libjli.so)
configure: error: The path given by --with-boot-jdk does not contain a valid Boot JDK
```

Comment (pshipton) [reply buildenv/docker/mkdocker.sh:379]: > What versions do we build in a docker container?

jdk8 and 11, as per the defaults.yml
```
    docker_image:
      8: 'eclipseopenj9/jenkins-agent-x86-centos6:24'
      11: 'eclipseopenj9/jenkins-agent-x86-centos6:24'
```

File: buildenv/docker/mkdocker.sh (hunk 7 above)
Comment (jdekonin) [reply buildenv/docker/mkdocker.sh]: updated

**Review by keithc-ca — APPROVED** [review]

**Pull request merged**